cli: `waypoint runner profile set` panics with no flags set

**Describe the bug**

If you run `waypoint runner profile set` with no flags set, the command panics due to the plugin-type flag being a string pointer.


```
brian@ubuntu:static λ waypoint runner profile set                                                 ±[main]
❌ Creating new runner profile
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x239360c]

goroutine 1 [running]:
github.com/hashicorp/waypoint/internal/cli.(*RunnerProfileSetCommand).Run(0x400031c580, {0x400047cf10, 0x0, 0x0})
        /media/psf/go/src/github.com/hashicorp/waypoint/internal/cli/runner_profile_set.go:237 +0x127c
github.com/mitchellh/cli.(*CLI).Run(0x4000cfe000)
        /media/psf/go/pkg/mod/github.com/mitchellh/cli@v1.1.2/cli.go:262 +0x4cc
github.com/hashicorp/waypoint/internal/cli.Main({0x4000124040?, 0x46190?, 0x25f1800?})
        /media/psf/go/src/github.com/hashicorp/waypoint/internal/cli/main.go:127 +0x528
main.main()
        /media/psf/go/src/github.com/hashicorp/waypoint/cmd/waypoint/main.go:14 +0xb4
```

**Steps to Reproduce**

`waypoint runner profile set`

**Expected behavior**

It should show the help text and explain what required flags are missing that are expected to be set.

**Waypoint Platform Versions**
Additional version and platform information to help triage the issue if
applicable:

* Waypoint CLI Version: 0.10.2
* Waypoint Server Platform and Version: (like `docker`, `nomad`, `kubernetes`)
* Waypoint Plugin: (like `aws/ecs`, `pack`, `azure`)

**Additional context**
Add any other context about the problem here.

--- a/.changelog/4013.txt
+++ b/.changelog/4013.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+cli: Fix panic in `waypoint runner profile set` when no flags are specified.
+```

--- a/internal/cli/runner_profile_set.go
+++ b/internal/cli/runner_profile_set.go
@@ -28,7 +28,7 @@ type RunnerProfileSetCommand struct {
 	flagOCIUrl             *string
 	flagEnvVar             map[string]string
 	flagEnvVars            []string
-	flagPluginType         *string
+	flagPluginType         string
 	flagPluginConfig       string
 	flagDefault            *bool
 	flagTargetRunnerAny    *bool
@@ -233,20 +233,20 @@ func (c *RunnerProfileSetCommand) Run(args []string) int {
 		}
 	}
 
-	if c.flagPluginType != nil || od.PluginType == "" {
-		if *c.flagPluginType == "" {
-			c.ui.Output(
-				"Flag '-plugin-type' must be set to a valid plugin type like 'docker' or 'kubernetes'.\n\n%s",
-				c.Help(),
-				terminal.WithErrorStyle(),
-			)
-			return 1
-		}
-		od.PluginType = *c.flagPluginType
-	}
+	if c.flagPluginType == "" && od.PluginType == "" {
+		c.ui.Output(
+			"Flag '-plugin-type' must be set to a valid plugin type like 'docker' or 'kubernetes'.\n\n%s",
+			c.Help(),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	} else if c.flagPluginType != "" {
+		// override plugin type from flag
+		od.PluginType = c.flagPluginType
+	} // else we stick with whats set originally in od
 
 	// Upsert
-	_, err := c.project.Client().UpsertOnDemandRunnerConfig(ctx, &pb.UpsertOnDemandRunnerConfigRequest{
+	resp, err := c.project.Client().UpsertOnDemandRunnerConfig(ctx, &pb.UpsertOnDemandRunnerConfigRequest{
 		Config: od,
 	})
 	if err != nil {
@@ -258,9 +258,9 @@ func (c *RunnerProfileSetCommand) Run(args []string) int {
 	}
 
 	if updated {
-		s.Update("Runner profile updated")
+		s.Update("Runner profile %q updated", resp.Config.Name)
 	} else {
-		s.Update("Runner profile created")
+		s.Update("Runner profile %q created", resp.Config.Name)
 	}
 	s.Done()
 
@@ -299,10 +299,11 @@ func (c *RunnerProfileSetCommand) Flags() *flag.Sets {
 			Usage:  "DEPRECATED. Please see `-env-var`.",
 		})
 
-		f.StringPtrVar(&flag.StringPtrVar{
-			Name:   "plugin-type",
-			Target: &c.flagPluginType,
-			Usage:  "The type of the plugin to launch for the on-demand runner, such as aws-ecs, kubernetes, etc.",
+		f.StringVar(&flag.StringVar{
+			Name:    "plugin-type",
+			Target:  &c.flagPluginType,
+			Default: "",
+			Usage:   "The type of the plugin to launch for the on-demand runner, such as aws-ecs, kubernetes, etc.",
 		})
 
 		f.StringVar(&flag.StringVar{

